### PR TITLE
[104] Fix NPE on null AQL expression

### LINKED ISSUE #104: Possible NPE during diagram refresh when a conditional style does not have a predicate
Rerported through [this forum post](https://www.eclipse.org/forums/index.php/m/1833946/):

```
Caused by: java.lang.NullPointerException: null
at org.eclipse.sirius.web.interpreter.ExpressionConverter.convertExpression(ExpressionConverter.java:57) ~[sirius-web-interpreter-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.interpreter.AQLInterpreter.evaluateExpression(AQLInterpreter.java:108) ~[sirius-web-interpreter-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.compat.diagrams.ContainerMappingStyleProvider.apply(ContainerMappingStyleProvider.java:54) ~[sirius-web-compatibility-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.compat.diagrams.ContainerMappingStyleProvider.apply(ContainerMappingStyleProvider.java:36) ~[sirius-web-compatibility-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.diagrams.components.NodeComponent.render(NodeComponent.java:70) ~[sirius-web-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderComponent(BaseRenderer.java:200) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderElement(BaseRenderer.java:121) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderChildren(BaseRenderer.java:170) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderInstance(BaseRenderer.java:143) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderElement(BaseRenderer.java:117) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderComponent(BaseRenderer.java:202) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderElement(BaseRenderer.java:121) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.diagrams.renderer.DiagramRenderer.render(DiagramRenderer.java:43) ~[sirius-web-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramService.create(DiagramService.java:71) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramRefreshManager.computeDiagram(DiagramRefreshManager.java:98) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramRefreshManager.refresh(DiagramRefreshManager.java:91) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramEventProcessor.refresh(DiagramEventProcessor.java:104) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.projects.ProjectEventProcessor.lambda$doHandle$3(ProjectEventProcessor.java:208) ~[sirius-web-spring-collaborative-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[na:na]
at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177) ~[na:na]
at java.base/java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3605) ~[na:na]
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[na:na]
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[na:na]
at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[na:na]
at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[na:na]
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497) ~[na:na]
at org.eclipse.sirius.web.spring.collaborative.projects.ProjectEventProcessor.doHandle(ProjectEventProcessor.java:207) ~[sirius-web-spring-collaborative-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.projects.ProjectEventProcessor.lambda$handle$1(ProjectEventProcessor.java:140) ~[sirius-web-spring-collaborative-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
```

### PR DESCRIPTION
**Pull request by pcdavid**
Fixes #104.

Signed-off-by: Pierre-Charles David <pierre-charles.david@obeo.fr>

### CODE REVIEW

File: backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/AQLInterpreterTestCases.java [hunk 1]
```diff
@@ -92,4 +92,20 @@ public void testBrokenEPackage() {
...
+    }
+
+    @Test
+    public void testEmptyExpressionEvaluatesToTrue() {
+        AQLInterpreter interpreter = new AQLInterpreter(List.of(), List.of(EcorePackage.eINSTANCE));
+        Result result = interpreter.evaluateExpression(Map.of(), ""); //$NON-NLS-1$
+        assertThat(result).isNotNull();
+        assertThat(result.asBoolean()).contains(Boolean.TRUE);
+    }
+
```
Comment (sbegaudeau) [inline backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/AQLInterpreterTestCases.java:110]: Just to be sure, is that the default behavior in Sirius RCP?

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
Changing this review to "Requested changes" since it appear that it does not match the expected behavior

File: backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/AQLInterpreterTestCases.java (hunk 1 above)
Comment (pcdavid) [reply backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/AQLInterpreterTestCases.java:110]: Good remark, it's actually the opposite. I'll change the new tests and update the actual code to match (hoping this does not have too many impacts).

Comment (pcdavid) [reply backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/AQLInterpreterTestCases.java:110]: Actually... I was right the first time :) I actually tested the behavior of Sirius Desktop, and either `aql:''` or `aql:'  '` (non-empty but blank strings) are true when internepreted as booleans. Our code in `org.eclipse.sirius.web.interpreter.Result.asBoolean()` it directly taken from https://github.com/pcdavid/org.eclipse.sirius/blob/master/plugins/org.eclipse.sirius.common/src/org/eclipse/sirius/common/tools/internal/interpreter/DefaultConverter.java#L36

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**